File: backend/app/routers/trainers.py
from fastapi import APIRouter, HTTPException

from app.data.mock_data import TRAINERS
from app.schemas import BookingRequest, BookingResponse, TrainerDetail, TrainerSummary

router = APIRouter(prefix="/api/trainers", tags=["trainers"])


@router.get("", response_model=list[TrainerSummary])
def list_trainers() -> list[dict]:
    return TRAINERS


@router.get("/{trainer_id}", response_model=TrainerDetail)
def get_trainer(trainer_id: str) -> dict:
    trainer = next((t for t in TRAINERS if t["id"] == trainer_id), None)
    if trainer is None:
        raise HTTPException(status_code=404, detail="Trainer not found")
    return trainer


@router.post("/{trainer_id}/book", response_model=BookingResponse)
def book_trainer(trainer_id: str, booking: BookingRequest) -> dict:
    trainer = next((t for t in TRAINERS if t["id"] == trainer_id), None)
    if trainer is None:
        raise HTTPException(status_code=404, detail="Trainer not found")
    kind = "групповое" if booking.type == "group" else "индивидуальное"
    return {"success": True, "message": f"Запись на {kind} занятие подтверждена! Ожидай SMS."}

File: frontend/lib/api/trainers.ts
import { apiRequest } from "./client";
import type { BookingRequest, BookingResponse, TrainerDetail, TrainerSummary } from "./types";

export function listTrainers(): Promise<TrainerSummary[]> {
  return apiRequest<TrainerSummary[]>("/api/trainers");
}

export function getTrainer(id: string): Promise<TrainerDetail> {
  return apiRequest<TrainerDetail>(`/api/trainers/${id}`);
}

export function bookTrainer(id: string, payload: BookingRequest): Promise<BookingResponse> {
  return apiRequest<BookingResponse>(`/api/trainers/${id}/book`, { method: "POST", body: payload });
}
